### PR TITLE
Autocomplete Autocomplete Slots

### PR DESCRIPTION
**Pull request by Austio**
Adds content distribution via dynamic slots for the autocomplete component.

To use the slot we provide some element that will hold the content for distribution.  That element will receive a `props` via a `scope` key that will be whatever we pass up from the child into the parent slot.

### Example Slot, props would be something like
`{item: { image_ur: 'foo.com', display: 'IMG' }}`

```
<template slot="item" scope="props">
  <img src={{props.item.image_url}} />
  <p>{{ props.item.display }}</p>
</template>
```

### Usage in a Parent Component
```
    <vg-autocomplete
      :items="[
        { display: 'January', value: '1'},
        { display: 'February', value: '2'},
        { display: 'March', value: '3'},
        ]"
      placeholder="Autocomplete"
    >
      <template slot="item" scope="props">
        I can be anything {{ props.item.display }}
      </template>
    </vg-autocomplete>
```

### Usage from the child providing props
```
<slot name="item" :item="item">
  {{ getDisplay(item) }}
</slot>
```

The content of the slot in the child is the default displayed, so we still get all of the sensible defaults for just displaying an array or an object as before.

Keep in mind that the content is compiled within whatever template it is located in, so to add a click handler to the slot you wrap the slot in the child and attach a handler similar to the examples in this repo.

### CODE REVIEW

File: src/components/autocomplete/index.vue [hunk 1]
```diff
@@ -13,21 +13,26 @@
...
       v-show="isActive"
-      :items="filteredItems"
-      @selected="handleSelect"
-      @mouseover="handleMenuMouseover"
-      :getDisplay="getDisplay"
-      :highlightedIndex="menuHighlightIndex"
+      class="vg-autocomplete-menu"
     >
-    </AutocompleteMenu>
+      <li
```
Comment (aurelienbottazini) [inline src/components/autocomplete/index.vue]: would it make sense to extract an autoCompleteItem component?

File: src/examples/autocomplete/index.vue [hunk 2]
```diff
@@ -8,6 +8,9 @@
         ]"
       placeholder="Autocomplete"
     >
+      <template slot="item" scope="props">
+        I can be anything {{ props.item.display }}
```
Comment (aurelienbottazini) [inline src/examples/autocomplete/index.vue]: very cool usage of scoped slots 👍 this is super cool! 🥇

File: src/components/autocomplete/index.vue (hunk 1 above)
Comment (aurelienbottazini) [inline src/components/autocomplete/index.vue]: Maybe simplify this component. I am not sure 🤔

Comment (Austio) [inline src/components/autocomplete/index.vue]: I initially had this as a menu item but to simplify the code I removed.  We would have to write this as a raw render function bc Vue is not as friendly with HOC using templates bc of the way they are compiled.

**Pull request merged**